micro_epsilon_scancontrol_driver/src/test.cpp

```diff
@@ -0,0 +1,163 @@
+#include <rclcpp/rclcpp.hpp>
+
+#include <llt.h>
+#include <mescan.h>
+#include <vector> 
+#include <micro_epsilon_scancontrol_msgs/srv/get_resolution.hpp>
+#include <micro_epsilon_scancontrol_msgs/srv/set_resolution.hpp>
+#include <functional>
+#include <iostream>
+
+
+#define MAX_INTERFACE_COUNT 5
+#define MAX_RESOLUTION 6
+
+
+guint32 resolution;
+std::vector<guint8> profile_buffer;
+TScannerType llt_type;
+guint32 profile_count, needed_profile_count;
+guint32 profile_data_size;
+
+EHANDLE* event;
+
+
+
+class TestNode : public rclcpp::Node{
+    public:
+        TestNode();
+        void getResolutionWrapper(const std::shared_ptr<micro_epsilon_scancontrol_msgs::srv::GetResolution::Request> request, std::shared_ptr<micro_epsilon_scancontrol_msgs::srv::GetResolution::Response> response);
+        void setResolutionWrapper(const std::shared_ptr<micro_epsilon_scancontrol_msgs::srv::SetResolution::Request> request, std::shared_ptr<micro_epsilon_scancontrol_msgs::srv::SetResolution::Response> response);
+        // void getResolutionWrapper();
+    private:
+        rclcpp::Service<micro_epsilon_scancontrol_msgs::srv::GetResolution>::SharedPtr service;
+        rclcpp::Service<micro_epsilon_scancontrol_msgs::srv::SetResolution>::SharedPtr set_resolution_srv;
+
+        std::unique_ptr<CInterfaceLLT> hLLT;
+
+};
+
+TestNode::TestNode():rclcpp::Node("test_node"){
+    RCLCPP_INFO(this->get_logger(), "test node started");
+
+    gint32 ret = 0;
+
+    char *interfaces[MAX_INTERFACE_COUNT];
```

Comment:
question: Why use a `char*` over a `string` or `string_view`?

Reply:
A part of this file was copied from a C code example. It shouldn't be in the commit, it was only for debugging a minimal example. I'll remove it!